### PR TITLE
MBS-11299 Fix report dir creation

### PR DESCRIPTION
**Pull request by sergioboi**
add validation when create the report dir

### CODE REVIEW

File: subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/DirectTestArtifactsProvider.kt [hunk 1]
```diff
@@ -29,7 +29,13 @@ internal class DirectTestArtifactsProvider(
 
     override fun generateFile(name: String, extension: String, create: Boolean): Result<File> {
         return provideReportDir().map { dir ->
-            File(dir, "$name.$extension")
+            val file = File(dir, "$name.$extension")
+            if (create && !file.exists()) {
```
Comment (sergioboi) [inline subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/DirectTestArtifactsProvider.kt:33]: `create` was ignored

File: subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/ReportDirProviderWithCreation.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    protected val runnerDirectory: File by lazy {
+        File(rootDir.value, "runner")
+    }
+
+    protected val testReportDir: File by lazy {
+        File(runnerDirectory, testDirGenerator.generateUniqueDir())
+    }
+}
+
+internal class ReportDirProviderForAdb(
```
Comment (sergioboi) [inline subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/ReportDirProviderWithCreation.kt:28]: When create provide report dir from runner we need only `path` without creation

File: subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/ReportDirProviderWithCreation.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+public interface ReportDirProvider {
+    public val reportDir: Result<File>
+}
+
+internal class SimpleDirProvider(reportDir: File) : ReportDirProvider {
+    override val reportDir: Result<File> = Result.Success(reportDir)
+}
+
+public abstract class ReportDirProviderByTest(
```
Comment (eugene-krivobokov) [inline subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/ReportDirProviderWithCreation.kt]: Can we make implementations internal?

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/ReportDirProviderWithCreation.kt (hunk 3 above)
Comment (sergioboi) [reply subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/ReportDirProviderWithCreation.kt]: Yes, my bad

**Pull request merged**